rfd/0058-oci-public-images.md

```diff
@@ -0,0 +1,208 @@
+
+#### **CloudFront**
+[CloudFront](https://aws.amazon.com/cloudfront/) will be leveraged to ensure fast, secure, downloads of Teleport OCI images across the globe.
+
+#### **Elastic Load Balancer**
+An internal [Amazon ELB](https://aws.amazon.com/elasticloadbalancing/) will be used to connect CloudFront CDN with the harbor process running within EKS. 
+
+#### **EKS**
+An Amazon EKS cluster will be deployed via IaC managed in the [cloud-terraform](https://github.com/gravitational/cloud-terraform/) repository. In order to reduce operational complexity, this cluster will exist within the `teleport-prod` AWS account alongside other resources. 
+
+An alternative to deploying this cluster within the `teleport-prod` AWS account would be to deploy this to an alternative AWS account and use cross-account IAM policies to grant access to the S3 bucket within `teleport-prod`. Deploying to a separate account would provide security isolation by account separation instead of through least-privileged IAM policies. 
+
+#### **Harbor**
+Harbor supports deployment through either `docker compose` or via `helm charts`. Harbor will be installed into a managed EKS cluster using helm as part of the implementation steps. Leveraging the Helm [terraform provider](https://registry.terraform.io/providers/hashicorp/helm/latest/docs), all changes to the Harbor installation will be managed with IaC. All changes will be peer-reviewed within the [cloud-terraform](https://github.com/gravitational/cloud-terraform/) repository.
+
+#### **S3**
+Harbor supports using S3 as a storage backend for images. Amazon S3 will be leveraged for this purpose. An S3 bucket will be created in the `teleport-prod` AWS account and will strictly follow standards as set in [Artifact Storage Standards](https://github.com/gravitational/cloud/blob/master/rfd/0017-artifact-storage-standards.md). The Harbor process should be the only service that needs access to modify this bucket. All container operations will occur through Harbor.
+
+### **Security Considerations**
+
+#### **Hosting**
+All resources laid out in this RFD will be hosted in the `teleport-prod` AWS account as defined by the [Artifact Storage Standards](https://github.com/gravitational/cloud/blob/master/rfd/0017-artifact-storage-standards.md) RFD. All resources will be configured and deployed using IaC with Terraform in the [cloud-terraform](https://github.com/gravitational/cloud-terraform/) repository.
+
+#### **Patching Harbor**
+Updates to the Harbor environment will be performed by the Core tooling and/or Release Engineers. Cloud tooling and/or Release Engineers can/will assist in this procedure as needed. 
```

Comment:
I expect this may be controversial -- core tooling has typically stayed out of the maintenance of self hosted software (drone being the exception -- and even then they're pushing towards GCB).  I'd run this by Jim, especially since the structure of the teams may be up in the air with my departure.

More generally, the discussion here cuts towards: Do we use ECR + some AWS config or Harbor/whatever-self-hosted + some AWS config.  My gut still says ECR, but you've spent a lot more time hands on with the tools than I have, so I trust your judgement.